### PR TITLE
Improve documentation and section split

### PR DESCRIPTION
**Pull request by nventuro**
Part of https://github.com/OpenZeppelin/docs.openzeppelin.com/issues/47

This fixes links across sections, improves their content according to recent feedback, and generally brings the documentation on par with the rest of our offering.

Note that the getting started links to a guide that doesn't yet exist (under review in https://github.com/OpenZeppelin/docs.openzeppelin.com/pull/48).

### CODE REVIEW

File: docs/modules/ROOT/pages/migrating-from-truffle.adoc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    transfer
+      when the recipient is not the zero address
+  ...
+```
+
+You will notice your tests start executing almost immediately: the https://docs.openzeppelin.com/contracts/2.x/[OpenZeppelin Contracts] have over 150 contracts and 2000 individual tests, and **testing setup time is under 2 seconds**.
+
+Enjoy lightning fast testing!
+
+NOTE: Unlike Truffle, Test Environment will not compile your contracts: so you will need to xref:getting-started.adoc#compiling[set this up yourself].
```
Comment (spalladino) [inline docs/modules/ROOT/pages/migrating-from-truffle.adoc]: I'd clarify here that you can compile with either OZ CLI or Truffle itself. If someone is migrating from truffle, I think it makes sense to tell them that they can keep their entire stack if they want to, and swap out only the testing part (we may even want to clarify this at the beginning of the guide).

File: docs/modules/ROOT/pages/getting-started.adoc [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+Test Environment requires your contracts to be compiled: you can use the https://docs.openzeppelin.com/sdk[OpenZeppelin CLI] to do this.
+
+```bash
+$ npm install --save-dev @openzeppelin/cli
+$ npx oz compile
+```
+
+Compilation artifacts will be stored in the `build/contracts` directory, where Test Environment (and most other tools) will read them from.
+
+TIP:
```
[the remark this answers is not in the document]
Comment (nventuro) [reply docs/modules/ROOT/pages/getting-started.adoc]: Heh, I forgot to put the add the actual suggestion (add `oz compile` to test script). Thanks!

File: docs/modules/ROOT/pages/getting-started.adoc [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+// test-environment.config.js
+
+module.exports = {
+  accounts: {
+    amount: 10, // Number of unlocked accounts
+    ether: 100, // Initial balance of unlocked accounts (in ether)
+  },
+
+  contracts: {
+    type: 'truffle', // Contract abstraction to use: 'truffle' for @truffle/contract or 'web3' for web3-eth-contract
```
[the remark this answers is not in the document]
Comment (nventuro) [reply docs/modules/ROOT/pages/getting-started.adoc]: They don't: `test-environment` brings all dependencies.

File: docs/modules/ROOT/pages/migrating-from-truffle.adoc (hunk 1 above)
Comment (nventuro) [reply docs/modules/ROOT/pages/migrating-from-truffle.adoc]: Great idea!

**Pull request merged**